packages/angular-cli/blueprints/class/index.ts

```diff
@@ -3,22 +3,22 @@ const dynamicPathParser = require('../../utilities/dynamic-path-parser');
 const Blueprint = require('../../ember-cli/lib/models/blueprint');
 const getFiles = Blueprint.prototype.files;
 
-module.exports = {
+export default Blueprint.extend({
   description: '',
 
   availableOptions: [
     { name: 'spec', type: Boolean }
   ],
 
-  normalizeEntityName: function (entityName) {
-    var parsedPath = dynamicPathParser(this.project, entityName.split('.')[0]);
+  normalizeEntityName: function (entityName: string) {
```

Comment:
There will be a second pass to add more type information and cleanup the code.
The goal here was to convert with minimal changes to reduce the chance of conflicts with other active PRs.